be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/PipelineError.scala

```diff
@@ -6,15 +6,17 @@ object ErrorCodes extends Enumeration {
   type ErrorCodes = Value
 
   // operation was successful (should never be used)
-  val Success: ErrorCodes = Value(0, "operation was successful")
+  val SUCCESS: ErrorCodes = Value(0, "operation was successful")
   // invalid action
-  val InvalidAction: ErrorCodes = Value(-1, "invalid action")
+  val INVALID_ACTION: ErrorCodes = Value(-1, "invalid action")
   // invalid resource (e.g. channel does not exist, channel was not subscribed to, etc.)
-  val InvalidResource: ErrorCodes = Value(-2, "invalid resource")
+  val INVALID_RESOURCE: ErrorCodes = Value(-2, "invalid resource")
   // resource already exists (e.g. lao already exists, channel already exists, etc.)
-  val AlreadyExists: ErrorCodes = Value(-3, "resource already exists")
+  val ALREADY_EXISTS: ErrorCodes = Value(-3, "resource already exists")
   // request data is invalid (e.g. message is invalid)
-  val InvalidData: ErrorCodes = Value(-4, "request data is invalid")
+  val INVALID_DATA: ErrorCodes = Value(-4, "request data is invalid")
   // access denied (e.g. subscribing to a “restricted” channel)
-  val AccessDenied: ErrorCodes = Value(-5, "access denied")
+  val ACCESS_DENIED: ErrorCodes = Value(-5, "access denied")
+  // server fault
+  val SERVER_FAULT: ErrorCodes = Value(-6, "server fault")
```

Comment:
I'm going to be super annoying and quote the specs nearly verbatim :-P

```suggestion
  // internal server error (e.g. crashed while processing, database unavailable, etc.)
  val SERVER_ERROR: ErrorCodes = Value(-6, "internal server error")
```